src/core/net/ip6.cpp

```diff
@@ -1271,6 +1271,14 @@ Error Ip6::HandleDatagram(Message &aMessage, Netif *aNetif, const void *aLinkMes
         hopLimit = header.GetHopLimit();
         aMessage.Write(Header::kHopLimitFieldOffset, hopLimit);
 
+#if OPENTHREAD_CONFIG_MULTI_RADIO
+        // Since the message will be forwarded, we clear the radio
+        // type on the message to allow the radio type for tx to be
+        // selected later (based on the radios supported by the next
+        // hop).
+        aMessage.ClearRadioType();
```

Comment:
nit: Do it just before `SuccessOrExit(error = Get<ThreadNetif>().SendMessage(aMessage));` ?